src/ripple/ledger/PaymentSandbox.h

```diff
@@ -188,6 +188,13 @@ class PaymentSandbox final
     apply (PaymentSandbox& to);
     /** @} */
 
+    // Return a map of balance changes on trust lines and amount of XRP destoryed.
+    // The low account is be the first account in the key.
+    // If the two accounts are equal, the map contains the total changes in currency
+    // regargless of issuer. This is useful to get the total change in XRP balances.
+    std::pair<std::map<std::tuple<AccountID, AccountID, Currency>, STAmount>, XRPAmount>
```

Comment:
@scottschurr @vinniefalco This function is used to find metadata differences between the two versions of payments. This data is only collected when a particular feature is enabled, and put into a `boost::optional`. My idea was if I collected additional metadata (offers changed, say), then I would add it to this one function.

I do think @scottschurr has a point about separating out the two methods, and will do so. However, when they are called they will likely be called from a lambda that returns a pair, so that the caller keeps a single `optional` around for all the collected metadata.